replicatedhq/ship
# Render templated messages in lifecycle for ship.yaml-driven installations.

What I Did
------------

Render templated messages in message endpoints for ship.yaml-driven installations.

How I Did it
------------

Check for `step.Message != nil` in `NavcycleRoutes.hydrateStep`, if set, use `step.Source.Message.Contents` as the template source, and set `step.Message.Contents` to the rendered value.

No new integration tests for this one since we don't really have a harness for testing cli.UI output in headless mode, and none test ship.yaml-driven UI workflows. This seems low risk enough that I'm okay with doing a little bit of manual testing for now.

How to verify it
------------

Write a ship.yaml with a `lifecycle.v1.message` that uses `{{repl Installation "customer_id"}}` or some other template string for the message. Ensure the message is properly rendered.

Description for the Changelog
------------

Render templated messages in lifecycle for ship.yaml-driven installations.


![By Lionel Walden - Christie's Los Angeles, Public Domain, https://commons.wikimedia.org/w/index.php?curid=12605929](https://upload.wikimedia.org/wikipedia/commons/e/ef/Lionel_Walden_-_Fishing_Along_the_Coast.jpg)
